### PR TITLE
Updated types for square-connect

### PR DESCRIPTION
**Pull request by Nican**
Added optional constructor information, ApiClient definition, and other minors changes.

Please fill in this template.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/square/connect-nodejs-sdk/blob/master/src/api/PaymentsApi.js https://github.com/square/connect-nodejs-sdk/blob/master/src/ApiClient.js
https://developer.squareup.com/reference/square/refunds-api/refund-payment
https://developer.squareup.com/reference/square/payments-api/get-payment

- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/square-connect/index.d.ts [hunk 1]
```diff
@@ -258,7 +258,11 @@ export enum CountryEnum {
     ZW = 'ZW',
 }
 
-export enum CurrencyEnum {
+/**
+ * Indicates the associated currency for an amount of money. Values correspond to [ISO
+ * 4217](https://wikipedia.org/wiki/ISO_4217).
+ */
+export enum Currency {
```
[the remark this answers is not in the document]
Comment (Nican) [reply types/square-connect/index.d.ts]: The currency enum is defined here: https://github.com/square/connect-nodejs-sdk/blob/master/src/index.js#L117 
Which refers to this file: https://github.com/square/connect-nodejs-sdk/blob/master/src/model/Currency.js

I just had to make `export enum Currency` match the same name as `var Currency = require('./model/Currency');`, and the enum seems to be working fine.

[the remark this answers is not in the document]
Comment (Nican) [reply types/square-connect/index.d.ts]: In relation to the `currency?: Currency | string;`, it makes it hard to set a value that was read from a database when it is not an enum. Same thing for storing the `status`. I am not a TypeScript export, so I might be doing something wrong. I will take a deeper look at it.

File: types/square-connect/index.d.ts [hunk 2]
```diff
@@ -10431,6 +10510,11 @@ export class MobileAuthorizationApi {
 }
 
 export class OAuthApi {
+    /**
+     * Constructs a new OAuthApi.
+     * @param apiClient Optional API client implementation to use, default to ApiClient.instance if unspecified.
+     */
+    constructor(apiClient?: ApiClient);
```
[the remark this answers is not in the document]
Comment (Nican) [reply types/square-connect/index.d.ts]: I posted a link to the official source code in the commit description. I literally just copied the constructor type and the comment. 
https://github.com/square/connect-nodejs-sdk/blob/master/src/api/PaymentsApi.js

File: types/square-connect/index.d.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for square-connect 2.20190814
+// Type definitions for square-connect 2.20190925
```
[the remark this answers is not in the document]
Comment (Nican) [reply types/square-connect/index.d.ts]: Ok. I will do that tonight when I am back home.

**Comment by Nican** [comment]
I am waiting on https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38917 before merging this.

**Comment by Nican** [comment]
Closed, as #38917 is already merging my changes.

**Comment by ddimitrioglo** [comment]
I picked your commit and added you as a reviewer for next updates.
Thanks a lot for contribution.
Hope that my PR will be merged soon...

**Pull request closed without merging**